modules/probe.sh

```diff
@@ -139,7 +139,18 @@ probe_kernel_image() {
     #    vmlinuz-3.10.0-862.el7.x86_64
     #    vmlinuz-4.15.0-20-generic
     #
-    kernel_image_path=$(sudo find ${1} -name ${image}* ! -name '*-rescue-*')
+    #Search for kernels, resolve symlinks, ignore invalid and select the latest if many
+    kernels=$(sudo find ${1} -name ${image}* ! -name '*-rescue-*' ! -name '*.old' \
+              -exec readlink -f {} \; | awk -F '/' '{print $NF,$0}' | sort -k1rV | uniq)
+    kernel_image_path=$(head -n 1 <<< "$kernels" | cut -f2- -d' ')
+    n=$(wc -l <<< "$kernels")
+
+    if [ "$n" -gt "1" ];then
+      log_info "$n kernels found, selecting newest: $kernel_image_path"
+      log_info "Set MENDER_GRUB_KERNEL_IMAGETYPE to override selected kernel"
```

Comment:
Would print the names of all the found kernels instead of just the number of kernel images found. This way the user can see what options are available if they want to override. e.g

```
log_warn "Found multiple kernel images: ${kernels}"
log_warn "Selecting newest kernel image: ${kernel_image_path}"
log_warn "Set MENDER_GRUB_KERNEL_IMAGETYPE to override selected kernel"
```

Note, this should be warning log level.